### PR TITLE
initial commit of careers

### PR DESCRIPTION
**Pull request by duallain**
Initial helm chart for careers
DB migration as preinstall hook
then generates a webdeployment, with service and nginx ingress to server the traffic
also creates a cronjob, to pull job data from greenhouse and push the static version of the website to s3 as well

### CODE REVIEW

File: charts/careers/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+1. check that the application deployed well by running the following commands
```
Comment (bkochendorfer) [inline charts/careers/templates/NOTES.txt:1]: Did you want to remove this or intentional to keep it in here?

File: charts/careers/templates/secrets.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+spec:
+  backendType: secretsManager
+  dataFrom:
+  - {{ .Values.externalSecrets.secretName }}
+{{- else }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "careers.fullname" . }}
+data:
```
Comment (bkochendorfer) [inline charts/careers/templates/secrets.yaml:28]: Do we want the secrets in here base64 encoded? I see it's not the production value just curious about this technique

File: charts/careers/templates/secrets.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  PG_NAME: Y2FyZWVycw==
+  PG_USER: Y2FyZWVycw==
+  PG_PASSWORD: ZGVmYXVsdHBhc3N3b3Jk
+  PG_HOST: Y2FyZWVycy1wc3Fs
+  PG_PORT: NTQzMg==
+  SECRET_KEY: dGhpcy1pcy1ub3QtdGhlLXByb2R1Y3Rpb24tdmFsdWU=
+  DEBUG: RmFsc2U=
+  ALLOWED_HOSTS: Kg==
+{{- end }}
+
```
Comment (bkochendorfer) [inline charts/careers/templates/secrets.yaml:55]:
```suggestion
```

File: charts/careers/templates/configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
```
Comment (bkochendorfer) [inline charts/careers/templates/configmap.yaml:1]:
```suggestion
---
apiVersion: v1
```

File: charts/careers/templates/hpa.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{{- if .Values.autoscaling.enabled }}
+apiVersion: autoscaling/v2beta1
```
Comment (bkochendorfer) [inline charts/careers/templates/hpa.yaml:2]:
```suggestion
---
apiVersion: autoscaling/v2beta1
```

File: charts/careers/templates/service.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
```
Comment (bkochendorfer) [inline charts/careers/templates/service.yaml:1]:
```suggestion
---
apiVersion: v1
```

File: charts/careers/templates/serviceaccount.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{{- if .Values.serviceAccount.create -}}
+apiVersion: v1
```
Comment (bkochendorfer) [inline charts/careers/templates/serviceaccount.yaml:2]:
```suggestion
---
apiVersion: v1
```

File: charts/careers/templates/web_deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,70 @@
+apiVersion: apps/v1
```
Comment (bkochendorfer) [inline charts/careers/templates/web_deployment.yaml:1]:
```suggestion
---
apiVersion: apps/v1
```

File: charts/careers/templates/periodic_job.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+apiVersion: batch/v1beta1
```
Comment (bkochendorfer) [inline charts/careers/templates/periodic_job.yaml:1]:
```suggestion
---
apiVersion: batch/v1beta1
```

**Review by bkochendorfer — COMMENTED** [review]
IDK how much we care about the `---` but I left some suggestions if we find that useful. I also noticed the `Chart.lock` was an empty file, is that intentional?

File: charts/careers/templates/NOTES.txt (hunk 1 above)
Comment (duallain) [reply charts/careers/templates/NOTES.txt:1]: I thought it'd be nice to leave in.  TBH I was on the fence.  Which way would you lean @bkochendorfer ?

File: charts/careers/templates/secrets.yaml (hunk 2 above)
Comment (duallain) [reply charts/careers/templates/secrets.yaml:28]: I was ok with it, like you said these aren't real values, but committing integration/functional test secrets to source code is weird for sure, but I personally like that pattern.

**Pull request merged**